clang/test/CheckedCRewriter/unsigned_signed_cast.c

```diff
@@ -0,0 +1,29 @@
+// RUN: cconv-standalone -alltypes %s -- | FileCheck -match-full-lines -check-prefixes="CHECK_ALL","CHECK" %s
+// RUN: cconv-standalone %s -- | FileCheck -match-full-lines -check-prefixes="CHECK_NOALL","CHECK" %s
+// RUN: cconv-standalone -output-postfix=checkedNOALL %s
+// RUN: %clang -c %S/unsigned_signed_cast.checkedNOALL.c
+// RUN: rm %S/unsigned_signed_cast.checkedNOALL.c
+
+/*unsigned integer cast*/
+
+unsigned int *foo = 0;
+int *bar = 0;  
+
+//CHECK: unsigned int *foo = 0;
+//CHECK: int *bar = 0;  
+
+/*unsigned characters cast*/
+
+unsigned char *yoo = 0; 
+char *yar = 0; 
+
+//CHECK: unsigned char *yoo = 0; 
+//CHECK: char *yar = 0; 
+
+/*C does not support unsigned floats, so we don't have to worry about that case*/
+
+/*ensure trivial conversion with parameter*/
+void test(int *x) {
+  foo = bar; 
```

Comment:
Can you explain why this assignment would prevent the pointer from being checked? Does the checked C compiler reject it? Are confident that it's doing so for a legitimate reason?

Frankly I don't see why treating a `int *` as an `unsigned int *` is unsafe.

Reply:
Yeah, it didn't seem to me that it would be unsafe either, but the checked c compiler rejects it if foo and bar are converted and yet this assignment occurs.
``` 
error: assigning to '_Ptr<unsigned int>' from incompatible type '_Ptr<int>'
```
I'm not sure if that's a legitimate enough reason since I've just sort of accepted the compiler as truth up to now. Are you saying this is more the checked c compiler's problem than ours?

Reply:
I checked and standard  `clang` emits a warning for the conversion, so they thought it was a bit iffy.

Reply:
OK, let's assume that Checked C clang knows what's its doing. Given that, this seems like the right fix.